### ISSUE TITLE
Incorrect implementation of tax when coupon has been applied

### ISSUE THREAD

**Issue by Nks** [bagisto/bagisto#10089]
**Describe the bug**
It looks like the system charges taxes and fees on the full product price and then deducts the discount from that total. 

Taxes and fees should only apply to the discounted amount: 
e.g.
- $100 subtotal on items
- $10 discount applied
- Amount Due = $90
- %Tax and Fee should be charged on $90 NOT $100.  

**To Reproduce**
1. Add product with the price 100$ and fixed amount coupon for 10$
2. Add product and apply coupon
3. It should apply tax for subtotal (90$), not to the total (100$)

**Expected behavior**
Taxes and fees should only apply to the discounted amount

**Screenshots**
![CleanShot 2024-08-27 at 18 05 06](https://github.com/user-attachments/assets/f4099171-309f-4e17-a9ce-2be9bb841d36)

**Additional context**
https://www.taxjar.com/blog/calculations/2021-12-sales-tax-discounts-coupons-promotions

> With few exceptions, sales tax applies to the gross sales of your products or taxable services. The specific regulations vary by state, but the sale price (or gross sales) will exclude any “discounts, including cash, term, or coupons that are not reimbursed by a third party that are allowed by a seller and taken by a purchaser on a sale.” (To take Georgia’s definition, for example.) In layman’s terms, that means if the original price of something you sell was $100, but you offer a 50% discount, then the taxable price is $50.

**Comment by devansh-webkul** [comment]
This is not in our roadmap. Feel free to customize this in your own fork.
